### PR TITLE
[Log] use default stderr logger if no raylog starting

### PR DESCRIPTION
**Pull request by ashione**
These changes of this PR are proposed to fix this issue https://github.com/ray-project/ray/issues/13741.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by raulchen** [comment]
Can you explain why this PR can fix the crash?

File: src/ray/util/logging.cc [hunk 1]
```diff
@@ -91,23 +102,32 @@ inline const char *ConstBasename(const char *filepath) {
   return base ? (base + 1) : filepath;
 }
 
+/// NOTE(lingxuan.zlx): Ray will put all of logging content into std err if no
+/// such logger given.
```
Comment (raulchen) [inline src/ray/util/logging.cc:106]:
```suggestion
/// A logger that prints logs to stderr.
/// This is the default logger if logging is not initialized.
```

**Review by raulchen — APPROVED** [review]

**Comment by ashione** [comment]
Rerun Travis CI

**Comment by ashione** [comment]
https://travis-ci.com/github/ray-project/ray/jobs/477733818

**Comment by ashione** [comment]
@simon-mo Java test suite has been relaunched 5 times and all passed. Could you double check it?

**Comment by ashione** [comment]
> Can you explain why this PR can fix the crash?

The previous solution will create a static stderr logger in first logging and reset format pattern at every log flush epoch, but it crashes in core worker deconstructor function when logger initialization (I have no idea why stderr logger would created in coreworker process exiting). So this PR make a singleton default stderr logger for log flushing if no default logger and no need to create the other logger.

**Comment by simon-mo** [comment]
```
+'[' 3 -gt 128 ']'
+find . -name 'hs_err_*log' -exec cat '{}' +
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f695e6ba3ae, pid=18633, tid=0x00007f696cbb9700
#
# JRE version: OpenJDK Runtime Environment (8.0_242-b08) (build 1.8.0_242-b08)
# Java VM: OpenJDK 64-Bit Server VM (25.242-b08 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libcore_worker_library_java.so+0x69e3ae]  spdlog::logger::flush_()+0x3e
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
---------------  T H R E A D  ---------------

```
https://travis-ci.com/github/ray-project/ray/jobs/477733818#L2910-L2927

**Comment by ashione** [comment]
> ```
> 
> +'[' 3 -gt 128 ']'
> 
> +find . -name 'hs_err_*log' -exec cat '{}' +
> 
> #
> 
> # A fatal error has been detected by the Java Runtime Environment:
> 
> #
> 
> #  SIGSEGV (0xb) at pc=0x00007f695e6ba3ae, pid=18633, tid=0x00007f696cbb9700
> 
> #
> 
> # JRE version: OpenJDK Runtime Environment (8.0_242-b08) (build 1.8.0_242-b08)
> 
> # Java VM: OpenJDK 64-Bit Server VM (25.242-b08 mixed mode linux-amd64 compressed oops)
> 
> # Problematic frame:
> 
> # C  [libcore_worker_library_java.so+0x69e3ae]  spdlog::logger::flush_()+0x3e
> 
> #
> 
> # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
> 
> #
> 
> # If you would like to submit a bug report, please visit:
> 
> #   http://bugreport.java.com/bugreport/crash.jsp
> 
> #
> 
> ---------------  T H R E A D  ---------------
> 
> 
> 
> ```
> 
> https://travis-ci.com/github/ray-project/ray/jobs/477733818#L2910-L2927

Thanks, it fails in last test of ten times retry. I'll check it first.

**Comment by ashione** [comment]
It's so weird to crash in core worker process deconstructor.

**Comment by ashione** [comment]
I guess the following explanation is make sense to me:
* static stderr logger in logger.cc is out of lifecyle scope of global variable std::unique_ptr<CoreWorkerProcess>
* Coreworkerprocess is still alive after stderrlogger deconstructed, so logging wil crash
@kfstorm @jovany-wang @raulchen

**Pull request merged**